src/generate/device.rs

```diff
@@ -191,12 +192,29 @@ pub fn render(d: &Device, config: &Config, device_x: &mut String) -> Result<Toke
             continue;
         }
 
-        out.extend(peripheral::render(
-            p,
-            &d.peripherals,
-            &d.default_register_properties,
-            config,
-        )?);
+        out.extend(
```

Comment:
Instead of using using the whole match expr,  do `out.extend(periph)` in the Ok branch

Dame thing gor every other occurance of this